### DISCUSSION TITLE
Using untrack to Store a function's value without Reactivity

### DISCUSSION THREAD

**Discussion by pecoram** [solidjs/solid#2347] [Q&A]
Hello, I would like to know if it's possible to use **untrack** to store the return value of a function, similar to how **createMemo** works, but without the **reactivity**.

```js
const untrackedValue = untrack(() => myFunction(props.id));
const memoValue = createMemo(() => myFunction(props.id));
```

My question is: can **untrack** be used to store the return value of myFunction so as to prevent it from getting recalled? What are the implications of using untrack in this way compared to createMemo?

Thank you!

**Comment by aniruddhaadak80** [comment]
Hey @pecoram ,

Yes, you can use `untrack` to store the return value of a function without reactivity. This prevents the function from being recalled when dependencies change. For example:

```javascript
const untrackedValue = untrack(() => myFunction(props.id));
```

Using `untrack` means that the value will remain static and won’t update if `props.id` changes. This is different from `createMemo`, which recalculates the value when dependencies change. Choose `untrack` for a one-time evaluation and `createMemo` for reactive updates.

Thank you .
